storage/store/inmemory/unsynchronized/registers.go

```diff
@@ -72,3 +75,18 @@ func (r *Registers) Store(registers flow.RegisterEntries, height uint64) error {
 
 	return nil
 }
+
+func (r *Registers) AddToBatch(batch storage.ReaderBatchWriter) error {
+	for id, value := range r.store {
+		encodedHeight := make([]byte, 8)
+		binary.BigEndian.PutUint64(encodedHeight, r.blockHeight)
+		key := append(encodedHeight, id.Bytes()...)
+
+		err := operation.UpsertByKey(batch.Writer(), key, value)
```

Comment:
this is different from the logic in the main implementation. is this the right way to do it @zhangchiqing?

Reply:
We are duplicating logs of persisting the registers, it would be error prone. 

We'd better just pass the RegisterEntries to the register stores for them to be stored. Refer to https://github.com/onflow/flow-go/blob/master/storage/pebble/registers.go#L111

I think the registers is stored in a separate database, right? 

So it's unnecessary to store registers with other execution data such as events, transaction results in the same batch. Note, batch can only be applied to a single database.